src/Microsoft.AspNetCore.Authentication.Facebook/FacebookMiddleware.cs

```diff
@@ -71,10 +71,30 @@ public class FacebookMiddleware : OAuthMiddleware<FacebookOptions>
             {
                 throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options.AppId)));
             }
+
             if (string.IsNullOrEmpty(Options.AppSecret))
             {
                 throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Exception_OptionMustBeProvided, nameof(Options.AppSecret)));
             }
+
+            if (Options.Scope.Count == 0)
```

Comment:
These should both be updated to use the new pattern from https://github.com/aspnet/Security/pull/680